rllib/agents/dqn/dqn.py

```diff
@@ -204,32 +225,27 @@ def execution_plan(
 
         def update_prio(item):
             samples, info_dict = item
-            if config.get("prioritized_replay"):
-                prio_dict = {}
-                for policy_id, info in info_dict.items():
-                    # TODO(sven): This is currently structured differently for
-                    #  torch/tf. Clean up these results/info dicts across
-                    #  policies (note: fixing this in torch_policy.py will
-                    #  break e.g. DDPPO!).
-                    td_error = info.get(
-                        "td_error", info[LEARNER_STATS_KEY].get("td_error")
+            prio_dict = {}
+            for policy_id, info in info_dict.items():
+                # TODO(sven): This is currently structured differently for
+                #  torch/tf. Clean up these results/info dicts across
+                #  policies (note: fixing this in torch_policy.py will
+                #  break e.g. DDPPO!).
+                td_error = info.get("td_error", info[LEARNER_STATS_KEY].get("td_error"))
+                samples.policy_batches[policy_id].set_get_interceptor(None)
+                batch_indices = samples.policy_batches[policy_id].get("batch_indexes")
+                # In case the buffer stores sequences, TD-error could
+                # already be calculated per sequence chunk.
+                if len(batch_indices) != len(td_error):
+                    T = local_replay_buffer.replay_sequence_length
+                    assert (
+                        len(batch_indices) > len(td_error)
+                        and len(batch_indices) % T == 0
                     )
-                    samples.policy_batches[policy_id].set_get_interceptor(None)
-                    batch_indices = samples.policy_batches[policy_id].get(
-                        "batch_indexes"
-                    )
-                    # In case the buffer stores sequences, TD-error could
-                    # already be calculated per sequence chunk.
-                    if len(batch_indices) != len(td_error):
-                        T = local_replay_buffer.replay_sequence_length
-                        assert (
-                            len(batch_indices) > len(td_error)
-                            and len(batch_indices) % T == 0
-                        )
-                        batch_indices = batch_indices.reshape([-1, T])[:, 0]
-                        assert len(batch_indices) == len(td_error)
-                    prio_dict[policy_id] = (batch_indices, td_error)
-                local_replay_buffer.update_priorities(prio_dict)
+                    batch_indices = batch_indices.reshape([-1, T])[:, 0]
+                    assert len(batch_indices) == len(td_error)
+                prio_dict[policy_id] = (batch_indices, td_error)
+            local_replay_buffer.update_priorities(prio_dict)
```

Comment:
How does PER sample using recurrent policies? Do you assign a single priority to the entire `replay_sequence`, or do you sum up the priorities of a `replay_sequence`?

Reply:
The entire sequence!
Today, Priority is is assigned per slot in the buffer.
Depending on the replayed item's unit (timestep, sequence, episode) you choose, one priority applies to one item.
Do you think we should change this? So far this replicates what we have done in the past.

Reply:
Yeah, we need to discuss restructuring our `agents` folder anyways. I'm actually in favor of separating every single algo, like APEX from DQN, and R2D2 from DQN (have them all as separate algos). APEX and R2D2 have no visibility right now, b/c they are buried inside DQN.